src/components/Button/Button.tsx

```diff
@@ -1,49 +1,186 @@
-import React, { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import * as React from 'react';
+import { Slot } from '@radix-ui/react-slot';
+import { cva, type VariantProps } from 'class-variance-authority';
 import { cn } from '../../helpers/classnames';
+import { Spinner } from '../spinner';
 
-export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
-  onClick?: () => void;
-  tooltip?: string;
-  filled?: boolean;
-  ghost?: boolean;
-  red?: true;
+const buttonVariants = cva(
+  [
+    'flex',
+    'font-medium',
+    'items-center',
+    'justify-center',
+    'gap-2',
+    'border',
+    'transition-colors',
+    'disabled:border-gray-400',
+    'disabled:bg-transparent',
+    'disabled:text-gray-400',
+    'disabled:active:opacity-100',
```

Comment:
add `'disabled:pointer-events-none',` to prevent hover/active effects on disabled